ext/sqlite3/sqlite3.c

```diff
@@ -1218,10 +1241,15 @@ PHP_METHOD(sqlite3, openBlob)
 
 	sqlite3_stream = emalloc(sizeof(php_stream_sqlite3_data));
 	sqlite3_stream->blob = blob;
+	sqlite3_stream->flags = flags;
 	sqlite3_stream->position = 0;
 	sqlite3_stream->size = sqlite3_blob_bytes(blob);
 
-	stream = php_stream_alloc(&php_stream_sqlite3_ops, sqlite3_stream, 0, "rb");
+	if (flags != 0) {
+		mode = "r+b";
+	}
```

Comment:
I'm just mirroring what SQLite is doing here: https://sqlite.org/c3ref/blob_open.html

> If the flags parameter is non-zero, then the BLOB is opened for read and write access. If the flags parameter is zero, the BLOB is opened for read-only access.

But I agree with you that it's not very readable, but I wasn't sure if we wanted to be closer to the sqlite3 lib or we wanted to change that to something more readable.